### PR TITLE
Update NewAzureSqlVM.cs to allow null license type

### PR DESCRIPTION
**Pull request by vasivara-MSFT**
Make LicenseType optional

## Description

Make LicenseType optional in powershell command. Sql server license type should be null in mooncake cloud. So we will set the license type as null and make the service throw an error depending on cloud

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/SqlVirtualMachine/SqlVirtualMachine/SqlVirtualMachine/Cmdlet/NewAzureSqlVM.cs [hunk 1]
```diff
@@ -75,11 +75,10 @@ public class NewAzureSqlVM : AzureSqlVMUpsertCmdletBase
         /// <summary>
         /// License type of new sql virtual machine
         /// </summary>
-        [Parameter(Mandatory = true,
+        [Parameter(Mandatory = false,
```
Comment (ranishan) [inline src/SqlVirtualMachine/SqlVirtualMachine/SqlVirtualMachine/Cmdlet/NewAzureSqlVM.cs:78]: You need to update the help documentation as well: New-AzSqlVM.md
https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md#updating-help-after-making-cmdlet-changes

File: src/SqlVirtualMachine/SqlVirtualMachine/SqlVirtualMachine/Cmdlet/NewAzureSqlVM.cs [hunk 2]
```diff
@@ -75,11 +75,10 @@ public class NewAzureSqlVM : AzureSqlVMUpsertCmdletBase
         /// <summary>
         /// License type of new sql virtual machine
         /// </summary>
-        [Parameter(Mandatory = true,
```
Comment (ranishan) [inline src/SqlVirtualMachine/SqlVirtualMachine/SqlVirtualMachine/Cmdlet/NewAzureSqlVM.cs:78]: So now can the New-AzSqlVMConfig command have the licenseType set to false as well. The command is to create a local SQLVM object that can be then used to create a New SqlVM.
PS C:\> $config = New-AzSqlVMConfig -LicenseType "PAYG"
PS C:\> New-AzSqlVM -ResourceGroupName "ResourceGroup01" -Name "vm" -SqlVM $config

**Review by VeryEarly — CHANGES_REQUESTED** [review]
- please fix test cases
- update changelog.md
- update help markdown

**Comment by msJinLei** [comment]
@vasivara-MSFT we are going to close the PR since no update for a long time. Feel free to submit a new one if you want to continue your work.

**Comment by isra-fel** [comment]
Closing the PR for lack of activity. Feel free to reopen to continue working on it. Thanks.

**Pull request closed without merging**